blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClient.scala

```diff
@@ -12,33 +12,31 @@ import scalaz.{-\/, \/-}
 object BlazeClient {
   def apply(manager: ConnectionManager, idleTimeout: Duration, requestTimeout: Duration): Client = {
     Client(Service.lift { req =>
-      def tryClient(client: BlazeClientStage, freshClient: Boolean, flushPrelude: Boolean): Task[DisposableResponse] = {
+      val key = RequestKey.fromRequest(req)
+      def tryClient(client: BlazeClientStage, flushPrelude: Boolean): Task[DisposableResponse] = {
         // Add the timeout stage to the pipeline
         val ts = new ClientTimeoutStage(idleTimeout, requestTimeout, bits.ClientTickWheel)
         client.spliceBefore(ts)
         ts.initialize()
 
         client.runRequest(req, flushPrelude).attempt.flatMap {
-          case \/-(r)    =>
+          case \/-(r)  =>
             val dispose = Task.delay {
-              if (!client.isClosed()) {
-                ts.removeStage
```

Comment:
The `reset()` is to reset the HTTP parser of the `BlazeClientStage` stage; this section is to eject the`ClientTimeoutStage` stage from the pipeline. Currently the `BlazeClientStage` resets itself [here](https://github.com/http4s/http4s/blob/master/blaze-client/src/main/scala/org/http4s/client/blaze/Http1ClientStage.scala#L201) when the body has been fully examined and if it still seems operational. 
I think the problem is the distinction between being `Idle` (required) for initiating a new request and being ['closed' ](https://github.com/http4s/http4s/blob/master/blaze-client/src/main/scala/org/http4s/client/blaze/Http1ClientStage.scala#L39) which just means the stage is in an `Error` state (which may be EOF). That said, I dont know how you would be attempting to recycle the connection without it either being in the idle or closed state considering that task should be evaluated after the one provided internally by the stage.